packages/flutter/test/material/popup_menu_theme_test.dart

```diff
@@ -0,0 +1,247 @@
+    expect(const PopupMenuThemeData().hashCode,
+        const PopupMenuThemeData().copyWith().hashCode);
+  });
+
+  test('PopupMenuThemeData null fields by default', () {
+    const PopupMenuThemeData popupMenuTheme =
+        PopupMenuThemeData();
+    expect(popupMenuTheme.color, null);
+    expect(popupMenuTheme.shape, null);
+    expect(popupMenuTheme.elevation, null);
+    expect(popupMenuTheme.textStyle, null);
+  });
+
+  testWidgets('Default PopupMenuThemeData debugFillProperties',
+      (WidgetTester tester) async {
+    final DiagnosticPropertiesBuilder builder = DiagnosticPropertiesBuilder();
+    const PopupMenuThemeData().debugFillProperties(builder);
+
+    final List<String> description = builder.properties
+        .where((DiagnosticsNode node) => !node.isFiltered(DiagnosticLevel.info))
+        .map((DiagnosticsNode node) => node.toString())
+        .toList();
+
+    expect(description, <String>[]);
+  });
+
+  testWidgets('PopupMenuThemeData implements debugFillProperties',
+      (WidgetTester tester) async {
+    final DiagnosticPropertiesBuilder builder = DiagnosticPropertiesBuilder();
+    PopupMenuThemeData(
+      color: const Color(0xFFFFFFFF),
+      shape: RoundedRectangleBorder(borderRadius: BorderRadius.circular(2.0)),
+      elevation: 2.0,
+      textStyle: const TextStyle(color: Color(0xffffffff)),
+    ).debugFillProperties(builder);
+
+    final List<String> description = builder.properties
+        .where((DiagnosticsNode node) => !node.isFiltered(DiagnosticLevel.info))
+        .map((DiagnosticsNode node) => node.toString())
+        .toList();
+
+    expect(description, <String>[
+      'color: Color(0xffffffff)',
+      'shape: RoundedRectangleBorder(BorderSide(Color(0xff000000), 0.0, BorderStyle.none), BorderRadius.circular(2.0))',
+      'elevation: 2.0',
+      'text style: TextStyle(inherit: true, color: Color(0xffffffff))'
+    ]);
+  });
+
+  testWidgets('Passing no PopupMenuThemeData returns defaults',
+      (WidgetTester tester) async {
+    final Key popupButtonKey = UniqueKey();
+    final Key popupButtonApp = UniqueKey();
+    final Key popupItemKey = UniqueKey();
+    final ThemeData theme = ThemeData();
+
+    await tester.pumpWidget(MaterialApp(
+      theme: theme,
+      key: popupButtonApp,
+      home: Material(
+        child: Column(
+          children: <Widget>[
+            PopupMenuButton<void>(
+              key: popupButtonKey,
+              itemBuilder: (BuildContext context) {
+                final List<PopupMenuEntry<Object>> list = <PopupMenuEntry<Object>>[
+                  PopupMenuItem<void>(
+                    key: popupItemKey,
+                    child: const Text(''),
+                  ),
+                ];
+                return list;
+              },
+            ),
+          ],
+        ),
+      ),
+    ));
+
+    await tester.tap(find.byKey(popupButtonKey));
+    await tester.pumpAndSettle();
+
+    final Material button = tester.widget<Material>(
```

Comment:
This formatting is a little hard to read. Do something like:
```
    final Material button = tester.widget<Material>(
      find.descendant(
        of: find.byKey(popupButtonApp),
        matching: find.byType(Material),
      ).last,
    );
```

Reply:
See [this section of the style guide](https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#indent-multi-line-argument-and-parameter-lists-by-2-characters) and the docs onwards for more guidance. Unfortunately, `dartfmt` doesn't exactly do what we want all the time.

Reply:
Done